concordium-std/src/impls.rs

```diff
@@ -556,6 +556,17 @@ impl<A, E: fmt::Debug> ExpectReport for Result<A, E> {
     }
 }
 
+impl<A, E: fmt::Debug> ExpectErrReport for Result<A, E> {
```

Comment:
I think what we want here is that A has Debug, and then in the error message for the Ok case display the value.

Reply:
And we don't need Debug on E.